slurm/datadog_checks/slurm/check.py

```diff
@@ -0,0 +1,444 @@
+        self.sinfo_collection_level = self.instance.get('sinfo_collection_level', 1)
+
+        # Binary paths
+        self.slurm_binaries_dir = self.init_config.get('slurm_binaries_dir', '/usr/bin/')
+
+        # CMD compilation
+        if self.collect_sinfo_stats:
+            self.sinfo_partition_cmd = self.get_slurm_command('sinfo', SINFO_PARTITION_PARAMS)
+            self.sinfo_collection_level = self.instance.get('sinfo_collection_level', 1)
+            if self.sinfo_collection_level > 1:
+                self.sinfo_node_cmd = self.get_slurm_command('sinfo', SINFO_NODE_PARAMS)
+                if self.sinfo_collection_level > 2:
+                    self.sinfo_node_cmd[-1] += SINFO_ADDITIONAL_NODE_PARAMS
+                if self.gpu_stats:
+                    self.sinfo_node_cmd[-1] += GPU_PARAMS
+            if self.gpu_stats:
+                self.sinfo_partition_cmd[-1] += GPU_PARAMS
+
+        if self.collect_squeue_stats:
+            self.squeue_cmd = self.get_slurm_command('squeue', SQUEUE_PARAMS)
+
+        if self.collect_sacct_stats:
+            self.sacct_cmd = self.get_slurm_command('sacct', SACCT_PARAMS)
+
+        if self.collect_sdiag_stats:
+            self.sdiag_cmd = self.get_slurm_command('sdiag', [])
+
+        if self.collect_sshare_stats:
+            self.sshare_cmd = self.get_slurm_command('sshare', SSHARE_PARAMS)
+
+        # Metric and Tag configuration
+        self.last_run_time = None
+        self.tags = self.instance.get('tags', [])
+
+        # Debug only. QOL feature for troubleshooting in the future. Allows me to dump specific
+        # debugging logs depending on what the issue is.
+        self.debug_sinfo_stats = is_affirmative(self.instance.get('debug_sinfo_stats', False))
+        self.debug_squeue_stats = is_affirmative(self.instance.get('debug_squeue_stats', False))
+        self.debug_sdiag_stats = is_affirmative(self.instance.get('debug_sdiag_stats', False))
+        self.debug_sshare_stats = is_affirmative(self.instance.get('debug_sshare_stats', False))
+        self.debug_sacct_stats = is_affirmative(self.instance.get('debug_sacct_stats', False))
+
+    def check(self, _):
+        self.collect_metadata()
+
+        commands = []
+
+        if self.collect_sinfo_stats:
+            commands.append(('sinfo', self.sinfo_partition_cmd, self.process_sinfo_partition))
+            if self.sinfo_collection_level > 1:
+                commands.append(('snode', self.sinfo_node_cmd, self.process_sinfo_node))
+
+        if self.collect_squeue_stats:
+            commands.append(('squeue', self.squeue_cmd, self.process_squeue))
+
+        if self.collect_sdiag_stats:
+            commands.append(('sdiag', self.sdiag_cmd, self.process_sdiag))
+
+        if self.collect_sshare_stats:
+            commands.append(('sshare', self.sshare_cmd, self.process_sshare))
+
+        if self.collect_sacct_stats and self.last_run_time is not None:
```

Comment:
Just so my understanding is correct, we don't run `collect_sacct_stats` the first time thorugh since `self.last_run_time` is `None`. We only run on consecutive runs one the `self.last_run_time` is given a timestamp.

Reply:
Yeah that's the idea. Because the sacct takes in a time filter that is like now -(how many seconds ago). I didn't want to use a static time and wanted it to be dynamic. So I opted to go for the 2 timestamp approach to get the delta. Kinda like how we do with silk events.

Also glad you found the comments helpful. I added a bunch because I know for sure I'm going to forget about this as soon as it's merged.